discovery-provider/alembic/versions/b0623220e904_create_notifications_table.py

```diff
@@ -68,6 +80,14 @@ def upgrade():
             info={"if_not_exists": True},
         )
 
+    if not unique_constraints_exists("notification", "uq_notification"):
+        op.create_unique_constraint(
+            "uq_notification",
+            "notification",
+            ["specifier", "group_id"],
```

Comment:
I suspect we'll want to make the index order `(group_id, specifier)` assuming we'll want to query on `group_id`?